app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,300 @@
+ * If `null` the warning icon will be shown at both ends.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun DismissedTabBackground(
+    dismissDirection: DismissDirection?,
+    isLastInCollection: Boolean,
+) {
+    Card(
+        modifier = Modifier.fillMaxSize(),
+        backgroundColor = FirefoxTheme.colors.layer3,
+        shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+        elevation = 0.dp,
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == StartToEnd) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == EndToStart) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+        }
+    }
+}
+
+/**
+ * Clips the Composable this applies to such that it cannot draw content / shadows outside it's top bound.
+ */
+private fun Modifier.clipTop() = this.then(
+    Modifier.drawWithContent {
+        val paddingPx = Constraints.Infinity.toFloat()
+        clipRect(
+            left = 0f - paddingPx,
+            top = 0f,
+            right = size.width + paddingPx,
+            bottom = size.height + paddingPx
+        ) {
+            this@drawWithContent.drawContent()
+        }
+    }
+)
+
+/**
+ * Get a friendlier short url for [url].
+ *
+ * @param url Full url to be shortened.
+ *
+ * @see toShortUrl
+ */
+@Composable
+private fun shortenUrl(url: String): String {
+    return if (inComposePreview) {
+        url
+    } else {
+        url.toShortUrl(LocalContext.current.components.publicSuffixList)
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun TabInCollectionDarkPreview() {
+    FirefoxTheme(theme = Theme.Dark) {
+        Column {
+            Box(modifier = Modifier.height(56.dp)) {
+                DismissedTabBackground(
+                    dismissDirection = StartToEnd,
+                    isLastInCollection = false,
+                )
+            }
```

Comment:
nit: missing a couple new line between Compose blocks in the previews

Reply:
Was intended on my part. Wanted to group in code what's shown in preview.